src/main/java/com/parsing/parsers/parserlot/ParserLotService.java

```diff
@@ -0,0 +1,118 @@
+package com.parsing.parsers.parserlot;
+
+import com.parsing.model.LotResult;
+import com.parsing.model.Status;
+import com.parsing.repository.LotResultRepository;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+
+@RequiredArgsConstructor
+@Slf4j
+@Service
+@Transactional(readOnly = true)
+public class ParserLotService {
+
+    private final LotResultRepository lotRepository;
+    List<String> urls = prepareURI();
+    private static final  String NOT_FOUND_HREF = "NOT FOUND HREF";
+
+    @Transactional
+    public void parse() throws IOException {
+        for (String url : urls) {
+            Document document = Jsoup.connect(url).get();
+            Element element = document.getElementsByClass("infobox-link").first();
+            if (element != null) {
```

Comment:
You can use Objects.nonNull(elements) method, it does the same but some people consider it more OOP or something, just information